src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborWriter.Array.cs

```diff
@@ -23,7 +23,22 @@ public void WriteStartArray(int definiteLength)
 
         public void WriteEndArray()
         {
+            if (!_remainingDataItems.HasValue)
+            {
+                // indefinite-length map, add break byte
+                EnsureWriteCapacity(1);
+                WriteInitialByte(new CborInitialByte(CborInitialByte.IndefiniteLengthBreakByte));
+            }
+
             PopDataItem(CborMajorType.Array);
         }
+
+        public void WriteStartArray()
```

Comment:
As the simpler method group this is probably what a caller would be more likely to use; but since indefinite feels like it could create compatibility problems with other implementations we should help them understand the cost.

I'd expect a zero argument version to be "I'll backfill the length for you" and the length one to be "here's the length".  Indefinite feels like it should be a different method group, like `WriteStartArrayIndefiniteLength`.

What's the plan for this method when conformance modes get added? Is it going to throw, or backfill?